src/ska_dict.rs

```diff
@@ -137,6 +166,7 @@ where
                     }
                 }
             }
+            iter_reads += 1;
```

Comment:
Just put this at the very top of the loop, then you don't need it in the if statement above too. Or it would be clearer to write as an if/else